src/vs/workbench/contrib/positronPlots/browser/components/plotsContainer.tsx

```diff
@@ -82,7 +87,49 @@ export const PlotsContainer = (props: PlotContainerProps) => {
 				plotHistory.scrollTop = plotHistory.scrollHeight;
 			}
 		}
-	});
+	}, [plotHistoryRef]);
+
+	useEffect(() => {
+		// Be defensive against null sizes when pane is invisible
+		if (plotWidth <= 0 || plotHeight <= 0) {
+			return;
+		}
+
+		const notify = () => {
+			let policy = props.positronPlotsService.selectedSizingPolicy;
+
+			if (policy instanceof PlotSizingPolicyIntrinsic) {
+				policy = new PlotSizingPolicyAuto;
+			}
+
+			const viewPortSize = {
+				height: plotHeight,
+				width: plotWidth,
+			}
+			let size = policy.getPlotSize(viewPortSize);
+			size = size ? size : viewPortSize;
+
+			props.positronPlotsService.setPlotsRenderSettings({
+				size,
+				pixel_ratio: DOM.getActiveWindow().devicePixelRatio,
```

Comment:
The device pixel ratio can vary between windows. We typically derive the window from a DOM element in the render tree (via `getWindow(element)`) instead of taking the active window for this reason. `getActiveWindow` pulls whichever window has focus which may not be the window in which the plot is rendering, and could have a different DPI.

Reply:
oh that makes sense, that should now be better